tools/prepare_prebuilt_workers/prepare_prebuilt_workers.go

```diff
@@ -0,0 +1,142 @@
+limitations under the License.
+*/
+
+// Configure is an executable that generates a defaults file for the manager.
+// It accepts a template file and replaces placeholders with data that may
+// change based on where the manager and container images will live and run.
+//
+// This tool uses Go's text/template package for templating, see
+// https://pkg.go.dev/text/template for a description of the syntax.
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os/exec"
+	"strings"
+)
+
+// Tests contains the values for fields that are accessible by
+// flags.
+type Tests struct {
+	preBuiltImagePrefix string
+	testTag             string
+	languagesToGitref   map[string]string
+}
+
+type langFlags []string
+
+func (l *langFlags) String() string {
+	var result string
+	for _, lang := range *l {
+		result = result + " " + lang
+	}
+	return result
+}
+
+func (l *langFlags) Set(value string) error {
+	*l = append(*l, value)
+	return nil
+}
+
+var languagesSelected langFlags
+
+func main() {
+	var test Tests
+
+	// specify the PREBUILD_IMAGE_PREFIX
+	flag.StringVar(&test.preBuiltImagePrefix, "p", "", "image registry to push the image")
+
+	// specify the tag of pre built image
+	flag.StringVar(&test.testTag, "t", "", "tags for pre-built images, this unique tag to identify the images build and pushed in current test, if a tag is not provided, will user default tag as: build-initiator-date-time")
+
+	// specify the languages wish to run tests with
+	flag.Var(&languagesSelected, "l", "languages and its GITREF wish to run tests, example: cxx:master")
+
+	flag.Parse()
+
+	if test.preBuiltImagePrefix == "" {
+		log.Fatalf("no registry provided, please provide a containr registry to store the images")
+	}
+
+	if test.testTag == "" {
+		log.Fatalf("no image tag provided, please provide a image tag")
+	} else if len(test.testTag) > 128 {
+		log.Fatalf("invalid tag name: A tag name may not start with a period or a dash and may contain a maximum of 128 characters.")
+	}
+
+	if len(languagesSelected) == 0 {
+		log.Fatalf("please select languages and the GITREF for tests you wish to run with pre-built images, for example: cxx:master")
+	}
+
+	test.languagesToGitref = map[string]string{}
+	for _, pair := range languagesSelected {
+		split := strings.Split(pair, ":")
+		if len(split) != 2 {
+			log.Fatalf("input error, please follow the format as language:gitref, for example: cxx:master")
+		}
+		test.languagesToGitref[split[0]] = split[1]
+	}
+
+	log.Print("selected language : GITREF")
+	fmt.Println(test.languagesToGitref)
+	formattedMap, _ := json.MarshalIndent(test.languagesToGitref, "", "  ")
+	log.Print(string(formattedMap))
+
+	for lang, gitRef := range test.languagesToGitref {
+		var containerRegistry = fmt.Sprintf("%s/%s:%s", test.preBuiltImagePrefix, lang, test.testTag)
+		log.Print(fmt.Sprintf("image registry: %s", containerRegistry))
+
+		// Build images
+		var dockerfileLocation = fmt.Sprintf("containers/pre_built_workers/%s/", lang)
```

Comment:
I think the Docker file location (or "root location") should not be hardcoded, but also passed in through a flag.